driver-core/src/main/com/mongodb/internal/TimeoutContext.java

```diff
@@ -158,4 +158,13 @@ public Timeout startServerSelectionTimeout() {
         long ms = getTimeoutSettings().getServerSelectionTimeoutMS();
         return StartTime.now().timeoutAfterOrInfiniteIfNegative(ms, MILLISECONDS);
     }
+
+    public Timeout startWaitQueueTimeout(final StartTime checkoutStart) {
+        final long ms = getTimeoutSettings().getMaxWaitTimeMS();
+        return checkoutStart.timeoutAfterOrInfiniteIfNegative(ms, MILLISECONDS);
+    }
+
+    public int getConnectTimeoutMs() {
+        return (int) getTimeoutSettings().getConnectTimeoutMS();
```

Comment:
This returns an int. I think it would be more consistent to return a Timeout, but it was used in a way where 0 indicated infinite timeout (rather than the usual negative is infinite, see above). I could not find what the user should use to indicate infinite connect timeout specified in our docs.

Reply:
This is because with Java sockets a timeout cannot be negative and 0 means infinite.  See: [socket.connect](https://docs.oracle.com/javase/8/docs/api/java/net/Socket.html#connect-java.net.SocketAddress-int-)

Reply:
Right, though I think that we should probably say the same in our own docs. (I didn't confirm that all paths used an API where 0 meant infinite, but I think this must be the case.)